test/utils/describeConformanceV5.js

```diff
@@ -28,13 +30,94 @@ function testComponentsProp(element, getOptions) {
   });
 }
 
+/**
+ * Material-UI theme has a components section that allows specifying default props, overrides and variants
+ * Components from @inheritComponent
+ * @param {React.ReactElement} element
+ * @param {() => ConformanceOptions} getOptions
+ */
+function testThemeComponents(element, getOptions) {
+  const render = createClientRender();
+
+  describe('theme: components', () => {
+    it("respect theme's defaultProps", () => {
+      const { muiName, testThemeComponentsDefaultPropWith: testProp = 'id' } = getOptions();
+      const theme = createMuiTheme({
+        components: {
+          [muiName]: {
+            defaultProps: {
+              [testProp]: 'testProp',
+            },
+          },
+        },
+      });
+
+      const { container } = render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
+
+      expect(container.firstChild[testProp]).to.equal('testProp');
+    });
+
+    it("respect theme's styleOverrides", () => {
+      const { muiName } = getOptions();
+
+      const testStyle = {
+        marginTop: '13px',
+      };
+
+      const theme = createMuiTheme({
+        components: {
+          [muiName]: {
+            styleOverrides: {
+              root: testStyle,
+            },
+          },
+        },
+      });
+
+      const { container } = render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
+
+      expect(container.firstChild).to.toHaveComputedStyle(testStyle);
+    });
+
+    it("respect theme's variants", () => {
+      const { muiName, testVariantProps = {} } = getOptions();
```

Comment:
We should also verify that these are in fact doing something by testing that the style isn't matched without the props. For example, render `[data-testid="with-props"]` and `[data-testid="without-props"]` and make sure only one matches the style.

Reply:
Done